examples/templates/docker-with-dotfiles/README.md

```diff
@@ -0,0 +1,38 @@
+---
+name: Develop in Docker with a dotfiles URL
+description: Run workspaces on a Docker host using registry images 
+tags: [local, docker]
+---
+
+# docker-with-dotfiles
+
+This is an example for deploying workspaces with a prompt for the users' dotfiles repo URI.
+
+## Getting started
+
+Run `coder templates init` and select this template. Follow the instructions that appear. 
+
+## How it works
+
+During workspace creation, Coder prompts you to specify a dotfiles URL via a Terraform variable. Once the workspace starts, the Coder agent runs `coder dotfiles` via the startup script:
+
+```hcl
+variable "dotfiles_uri" {
+  description = <<-EOF
+  Dotfiles repo URI (optional)
+
+  see https://dotfiles.github.io
+  EOF
+    # The codercom/enterprise-* images are only built for amd64
+  default = ""
+}
+
+resource "coder_agent" "dev" {
+  ...
+  startup_script = var.dotfiles_uri != "" ? "/tmp/tmp.coder*/coder dotfiles -y ${var.dotfiles_uri}" : null
```

Comment:
While `coder` was in PATH, it didn't seem to work in the startup script. Even when I specified /bin/bash. Any ideas on how to use the agent's shell?

I'm also on a slightly older build. I'll try upgrading and see if this fixes it

Reply:
The problem is that the PATH is not set _for_ the agent.

One option is to set the `PATH` earlier, before agent start, e.g. here: https://github.com/coder/coder/blob/d0ed107b08f83d7b5c1f4794937e16adda0137b1/provisionersdk/agent.go#L35

Something like `export PATH=${PATH}:${BINARY_LOCATION}`.

**OR** we can add it to the path for the startup script here:

https://github.com/coder/coder/blob/b65259f95e53e96cd8daf21a5c3de46e51a56f85/agent/agent.go#L176

Like we do here:

https://github.com/coder/coder/blob/b65259f95e53e96cd8daf21a5c3de46e51a56f85/agent/agent.go#L351